persistence/album_repository_test.go

```diff
@@ -86,4 +90,51 @@ var _ = Describe("AlbumRepository", func() {
 			Expect(getMinYear("2000 0 1800")).To(Equal(1800))
 		})
 	})
+
+	Describe("getCoverFromPath", func() {
+		testFolder, _ := ioutil.TempDir("", "album_persistence_tests")
+		if err := os.MkdirAll(testFolder, 0777); err != nil {
+			panic(err)
+		}
+		if _, err := os.Create(filepath.Join(testFolder, "Cover.jpeg")); err != nil {
+			panic(err)
+		}
+		if _, err := os.Create(filepath.Join(testFolder, "FRONT.PNG")); err != nil {
+			panic(err)
+		}
+
+		testPath := filepath.Join(testFolder, "somefile.test")
+		It("returns audio file for embedded cover", func() {
+			conf.Server.CoverArtPriority = "embedded, cover.*, front.*"
+			Expect(getCoverFromPath(testPath, true)).To(Equal(""))
+		})
+
+		It("returns external file when no embedded cover exists", func() {
+			conf.Server.CoverArtPriority = "embedded, cover.*, front.*"
+			Expect(getCoverFromPath(testPath, false)).To(Equal(filepath.Join(testFolder, "Cover.jpeg")))
```

Comment:
if `Cover.jpeg` matches `cover.*`, it will fail in the `cover.go`'s `readFromFile` function if running on Linux. Maybe it should be case-sensitive?